haystack/document_store/weaviate.py

```diff
@@ -0,0 +1,592 @@
+            username: str = None,
+            password: str = None,
+            index: str = "Document",
+            vector_dim: int = 768,
+            text_field: str = "text",
+            name_field: str = "name",
+            faq_question_field = "question",
+            meta_field = "meta",
+            similarity: str = "dot_product",
+            index_type: str = "hnsw",
+            custom_schema: Optional[dict] = None,
+            module_name: str = "text2vec-transformers",
+            update_existing_documents: bool = False,
+            return_embedding: bool = False,
+            embedding_field: str = "embedding",
+            progress_bar: bool = True,
+            **kwargs,
+    ):
+        """
+        :param weaviate_url: Weaviate server connection URL for storing and processing documents and vectors.
+                             For more details, refer "https://www.semi.technology/developers/weaviate/current/getting-started/installation.html"
+        :param timeout_config: Weaviate Timeout config as a tuple of (retries, time out seconds).
+        :param username: username (standard authentication via http_auth)
+        :param password: password (standard authentication via http_auth)
+        :param index: Index name for document text, embedding and metadata (in Weaviate terminology, this is a "Class" in Weaviate schema).
+        :param vector_dim: The embedding vector size. Default: 768.
+        :param text_field: Name of field that might contain the answer and will therefore be passed to the Reader Model (e.g. "full_text").
+                           If no Reader is used (e.g. in FAQ-Style QA) the plain content of this field will just be returned.
+        :param name_field: Name of field that contains the title of the the doc
+        :param faq_question_field: Name of field containing the question in case of FAQ-Style QA
+        :param meta_field : Name of field to store all the meta data (key value pairs) in the Document object
+        :param similarity: The similarity function used to compare document vectors. 'dot_product' is the default.
+        :param index_type: Index type of any vector object defined in weaviate schema. The vector index type is pluggable.
+                           Currently, HSNW is only supported.
+                           See: https://www.semi.technology/developers/weaviate/current/more-resources/performance.html
+        :param custom_schema: Allows to create custom schema in Weaviate, for more details
+                           See https://www.semi.technology/developers/weaviate/current/data-schema/schema-configuration.html
+        :param module_name : Vectorization module to convert data into vectors. Default is "text2vec-trasnformers"
+                            For more details, See https://www.semi.technology/developers/weaviate/current/modules/
+        :param update_existing_documents: Whether to update any existing documents with the same ID when adding
```

Comment:
As you saw in https://github.com/deepset-ai/haystack/issues/1069, this arg is going to change soon.